osu.Game/Overlays/MusicController.cs

```diff
@@ -136,11 +136,20 @@ public bool TogglePause()
         }
 
         /// <summary>
-        /// Play the previous track.
+        /// Play the previous track or restart the current track if it's current time below 5000ms
         /// </summary>
         /// <returns>Whether the operation was successful.</returns>
         public bool PrevTrack()
         {
+            var currentTrackPosition = current?.Track.CurrentTime;
+
+            if (currentTrackPosition >= 5000)
```

Comment:
the number of milliseconds to allow playing the previous track should probably be a `double` constant.

Reply:
Yeah, I'd extract something like a `RestartCutoffPoint` constant here. Additionally this check should not be duplicated verbatim twice.

I'd extract a private method at the least (`bool ShouldReset(Track track)`), or, more controversially, introduce a tri-value inner enum:
```csharp
enum PrevActionResult
{
    None,
    Restart,
    Previous
}
```
and switch on that in `OnPressed` to get the appropriate message.